### PR TITLE
Feature/query microservices

### PR DESCRIPTION
**Pull request by jwomeara**

### CODE REVIEW

File: api/src/main/java/datawave/webservice/model/Model.java [hunk 1]
```diff
@@ -11,12 +11,17 @@
...
+
+import com.google.common.collect.HashMultimap;
+import com.google.common.collect.Multimap;
+
 import datawave.query.model.FieldMapping;
 import datawave.webservice.HtmlProvider;
 import datawave.webservice.result.BaseResponse;
-import lombok.Data;
 
-@Data
```
Comment (jwomeara) [inline api/src/main/java/datawave/webservice/model/Model.java:19]: Apparently I removed lombok...  thoughts?  opinions?

File: service/src/main/java/datawave/microservice/model/ModelController.java [hunk 2]
```diff
@@ -48,7 +48,7 @@
                                 url = "https://github.com/NationalSecurityAgency/datawave-dictionary-service"))
 @Slf4j
 @RestController
-@RequestMapping(path = "/model",
+@RequestMapping(path = "/model/v1",
```
Comment (jwomeara) [inline service/src/main/java/datawave/microservice/model/ModelController.java:51]: This is a change to one of our external interfaces.  Are we ok with that?

**Review by jwomeara — COMMENTED** [review]
Notable changes in this PR:
 - Major version bumps.
 - Model class moved here (and i think removed from the main datawave project)
 - /v1 path added to ModelController

File: api/src/main/java/datawave/webservice/model/Model.java (hunk 1 above)
Comment (apmoriarty) [reply api/src/main/java/datawave/webservice/model/Model.java:19]: Seems like it was unused, so probably fine. If we wanted to use lombok more places that would be fine too

Comment (matthpeterson) [reply api/src/main/java/datawave/webservice/model/Model.java:19]: I have a mild preference for less code with lmbok, but there are valid arguments for and against.

File: service/src/main/java/datawave/microservice/model/ModelController.java (hunk 2 above)
Comment (matthpeterson) [reply service/src/main/java/datawave/microservice/model/ModelController.java:51]: Is the plan to provide support for multiple versions at once?  I like the versioned path approach.

Perhaps the path without a version could remain and map to the latest non-experimental version.

**Review by matthpeterson — APPROVED** [review]

File: api/src/main/java/datawave/webservice/model/Model.java (hunk 1 above)
Comment (alerman) [reply api/src/main/java/datawave/webservice/model/Model.java:19]: I have a strong preference for lombok, and brought it up some places in other PR Reviews. I use it up and down on other projects and have never had any issues.

File: api/src/main/java/datawave/webservice/model/Model.java [hunk 3]
```diff
@@ -40,15 +45,53 @@ public Model(String jqueryUri, String datatablesUri) {
...
     
     @XmlAttribute(name = "name", required = true)
-    private String name;
+    private String name = null;
     
     @XmlElementWrapper(name = "Mappings")
     @XmlElement(name = "Mapping")
     private TreeSet<FieldMapping> fields = new TreeSet<FieldMapping>();
     
+    public String getName() {
```
Comment (alerman) [inline api/src/main/java/datawave/webservice/model/Model.java:54]: This entire block could be replaced using lombok.
`@Data
@EqualsAndHashcode
@ToString`

File: service/src/main/java/datawave/microservice/model/ModelController.java (hunk 2 above)
Comment (jwomeara) [reply service/src/main/java/datawave/microservice/model/ModelController.java:51]: typically for our endpoints on the other services we put the version designator before the endpoint.  i.e. `/v1/model`

@ivakegg didn't you make this change?  what are your thoughts?

File: api/src/main/java/datawave/webservice/model/Model.java (hunk 3 above)
Comment (jwomeara) [reply api/src/main/java/datawave/webservice/model/Model.java:54]: yeah... it probably could, but do we want lombok chocolate in our query microservices peanut butter?

File: api/src/main/java/datawave/webservice/model/Model.java (hunk 1 above)
Comment (jwomeara) [reply api/src/main/java/datawave/webservice/model/Model.java:19]: there are already a lot of changes/PRs in this effort.  should the scope expand to include lombok, or should we make a ticket and save that for a rainy day?  i am in favor of the latter.

Comment (matthpeterson) [reply api/src/main/java/datawave/webservice/model/Model.java:19]: Let's minimize the scope of these changes.

**Review by apmoriarty — APPROVED** [review]

File: service/src/main/java/datawave/microservice/model/ModelController.java (hunk 2 above)
Comment (ivakegg) [reply service/src/main/java/datawave/microservice/model/ModelController.java:51]: For this one, I think we should keep is as dictionary/model/v1 to be consistent with dictionary/edge/v1 and dictionary/data/v1.  I expect we might change the interface on one of those and not the others as opposed to other applications which have primarily one main controller.

Comment (jwomeara) [reply service/src/main/java/datawave/microservice/model/ModelController.java:51]: Per our earlier conversation... yes, i agree.

**Pull request merged**